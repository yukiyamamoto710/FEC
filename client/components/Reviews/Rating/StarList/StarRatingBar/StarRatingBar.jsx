import React from 'react';
import PropTypes from 'prop-types';

const StarRatingBar = (props) => {
  const { per } = props;
  let bar = {
    border: '5px solid rgb(102, 162, 182)',
    width: `${per}%`,
    backgroundColor: 'rgb(102, 162, 182)',
  };
  if (per === 0) {
    bar = {
      display: 'none',
    };
  }

  return (
    <div
      data-testid="starRatingBar"
      className="starRatingBarBase"
    >
      <div style={bar} />
    </div>
  );
};

StarRatingBar.propTypes = {
  per: PropTypes.number,
};

StarRatingBar.defaultProps = {
  per: 0,
};

export default StarRatingBar;
