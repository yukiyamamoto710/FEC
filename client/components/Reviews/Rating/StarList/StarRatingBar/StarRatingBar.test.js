/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  cleanup,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import StarRatingBar from './StarRatingBar';

afterEach(cleanup);

it('should render', () => {
  const { getByTestId } = render(<StarRatingBar per={0} />);
  expect(getByTestId('starRatingBar')).toBeInTheDocument();
});
it('matches snapShot', () => {
  const tree = renderer.create(<StarRatingBar per={0} />).toJSON();
  expect(tree).toMatchSnapshot();
});
