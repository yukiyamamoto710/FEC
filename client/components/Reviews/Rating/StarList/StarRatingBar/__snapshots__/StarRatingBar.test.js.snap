// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<div
  className="starRatingBarBase"
  data-testid="starRatingBar"
>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  />
</div>
`;
